libraries/AP_BattMonitor/AP_BattMonitor_TIBQ76952.cpp

```diff
@@ -0,0 +1,937 @@
+    // COV (over-voltage), CUV (under-voltage)
+    set_register(TIBQ769x2_EnabledProtectionsA, 0xBC, 1);
+#endif
+
+#if defined(DISABLE_PROTECTION_B)
+    set_register(TIBQ769x2_EnabledProtectionsB, 0x00, 1);
+#else
+    // Enable all protections in 'Enabled Protections B' 0x9262 = 0xF7
+    // Enables OTF (over-temperature FET), OTINT (internal over-temperature), OTD (over-temperature in discharge),
+    // OTC (over-temperature in charge), UTINT (internal under-temperature), UTD (under-temperature in discharge), UTC (under-temperature in charge)
+    set_register(TIBQ769x2_EnabledProtectionsB, 0xF7, 1);
+#endif
+
+    // 'Default Alarm Mask' - 0x926D Enables the FullScan and ADScan bits, default value = 0xF800
+    set_register(TIBQ769x2_DefaultAlarmMask, 0xF882, 2);
+
+    // Set up Cell Balancing Configuration - 0x9335 = 0x03 - Automated balancing while in Relax or Charge modes
+    set_register(TIBQ769x2_BalancingConfiguration, 0x03, 1);
+
+    // Set up CUV (under-voltage) Threshold - 0x9275 = 0x31 (2479 mV)
+    // CUV Threshold is this value multiplied by 50.6mV
+    set_register(TIBQ769x2_CUVThreshold, 0x31, 1);
+
+    // Set up COV (over-voltage) Threshold - 0x9278 = 0x55 (4301 mV)
+    // COV Threshold is this value multiplied by 50.6mV
+    set_register(TIBQ769x2_COVThreshold, 0x55, 1);
+
+    // Set up OCC (over-current in charge) Threshold - 0x9280 = 0x05 (10 mV = 10A across 1mOhm sense resistor) Units in 2mV
+    set_register(TIBQ769x2_OCCThreshold, 0x05, 1);
+
+    // Set up OCD1 Threshold - 0x9282 = 0x0A (20 mV = 20A across 1mOhm sense resistor) units of 2mV
+    set_register(TIBQ769x2_OCD1Threshold, 0x0A, 1);
+
+    // Set up SCD Threshold - 0x9286 = 0x05 (100 mV = 100A across 1mOhm sense resistor) 0x05=100mV
+    set_register(TIBQ769x2_SCDThreshold, 0x05, 1);
+
+    // Set up SCD Delay - 0x9287 = 0x03 (30 us) Enabled with a delay of (value - 1) * 15 μs; min value of 1
+    set_register(TIBQ769x2_SCDDelay, 0x03, 1);
+
+    // Set up SCDL Latch Limit to 1 to set SCD recovery only with load removal 0x9295 = 0x01
+    // If this is not set, then SCD will recover based on time (SCD Recovery Time parameter).
+    set_register(TIBQ769x2_SCDLLatchLimit, 0x01, 1);
+
+    // Exit CONFIGUPDATE mode - Subcommand 0x0092
+    sub_command(TIBQ769x2_EXIT_CFGUPDATE);
+
+    sub_command(TIBQ769x2_ALL_FETS_OFF);
+    sub_command(TIBQ769x2_FET_ENABLE);
+
+    // Mark configuration as complete to prevent repeated attempts
+    configured = true;
+    Debug("BQ76952: Configuration complete");
+
+    // report success
+    return true;
+}
+
+// read battery voltage, current and temperature
+void AP_BattMonitor_TIBQ76952::read_voltage_current_temperature()
+{
+    // exit immediately if full voltage scan has not completed (FULL_SCAN bit 7)
+    uint16_t alarm_raw_status = direct_command_read_2bytes(TIBQ769x2_AlarmRawStatus);
+    if (!BIT_IS_SET(alarm_raw_status, 7)) {
```

Comment:
Please name this bit

Reply:
ok, safety status bit definitions added and used